### PR TITLE
New approach in ConstrainedIBC

### PR DESCRIPTION
**Pull request by yashcodes**

### CODE REVIEW

File: src/ConstrainedIBC.jl [hunk 1]
```diff
@@ -37,16 +37,19 @@ function contraction(f::Function, C, global_min::Float64, X::IntervalBox{N,T}, c
...
+                break
+            end
+            if j == length(constraints)
+                return (true, point)
+            end
         end
     end
-
-    return point
+    return (false)
```
Comment (dpsanders) [inline src/ConstrainedIBC.jl]: This function is type unstable. Try returning the same type always - it should be much faster (benchmark this).

File: src/ConstrainedIBC.jl [hunk 2]
```diff
@@ -37,16 +37,19 @@ function contraction(f::Function, C, global_min::Float64, X::IntervalBox{N,T}, c
...
-
-    for j in 1:length(constraints)
-        if !(invokelatest(constraints[j].C, point) ⊆ constraints[j].bound)
-            point = generate_random_feasible_point(X, constraints)
-            break
+    for i in 1:30
+        point = [X[j].lo + (1-rand())*(X[j].hi - X[j].lo) for j in 1:length(X)]      # discover a random point in interval box X
+        for j in 1:length(constraints)
+            if !(invokelatest(constraints[j].C, point) ⊆ constraints[j].bound)
+                point = generate_random_feasible_point(X, constraints)
```
Comment (dpsanders) [inline src/ConstrainedIBC.jl]: This still has an infinite regress?

Comment (yashcodes) [reply src/ConstrainedIBC.jl]: `point= generate_random_feasible_point(X, constraints)` 
should not be there, thats a mistake

File: src/ConstrainedIBC.jl (hunk 1 above)
Comment (yashcodes) [reply src/ConstrainedIBC.jl]: OK

File: src/ConstrainedIBC.jl [hunk 3]
```diff
@@ -93,36 +98,50 @@ function ibc_minimise(f::Function , X::IntervalBox{N,T}, constraints::Vector{Con
...
-            x_best = SVector(mid(X))
-            if diffevol_chnl != Nothing
-                if debug
-                    println("Box send to DifferentialEvolution: ", x_best )
-                end
-                if info.iterations % 200 == 0
-                    put!(diffevol_chnl, (x_best, global_min, X))  # sending best individual to diffevol
-                else
-                    put!(diffevol_chnl, (x_best, global_min, nothing))
+        output = generate_random_feasible_point(X, constraints)   # finding a feasible point in the interval box if there present any
```
Comment (dpsanders) [inline src/ConstrainedIBC.jl]: You should do something like

status, output =....

Comment (yashcodes) [reply src/ConstrainedIBC.jl]: Yes right it would be nicer that way.

**Comment by yashcodes** [comment]
But the results are not getting any right with this approach too :(

**Comment by dpsanders** [comment]
My guess as to why the code is not working is that you are removing constraints.
As a first step I suggest never modifying the list of constraints.

**Comment by yashcodes** [comment]
Ok lets try that too.

**Pull request merged**